plugins/modules/katello_content_view.py

```diff
@@ -95,106 +98,113 @@
-    content_view_dict = sanitize_entity_dict(entity_dict, name_map)
+    entity = module.find_resource_by_name('content_views', name=entity_dict['name'], params=scope, failsafe=True)
 
-    changed, content_view_entity = naildown_entity(ContentView, content_view_dict, content_view_entity, state, module)
+    changed, content_view_entity = module.ensure_entity('content_views', entity_dict, entity, params=scope)
 
     # only update CVC's of newly created or updated CCV's
-    if state == 'present' or (state == 'present_with_defaults' and changed):
-        current_cvcs = []
-        if hasattr(content_view_entity, 'content_view_component'):
-            current_cvcs = [cvc.read() for cvc in content_view_entity.content_view_component]
-        if 'components' in entity_dict and content_view_entity.composite:
-            for component in entity_dict['components']:
-                cvc = component.copy()
-                cvc['content_view'] = find_content_view(module, name=component['content_view'], organization=entity_dict['organization'])
-                cvc_matched = None
-                for _cvc in current_cvcs:
-                    if _cvc.content_view.id == cvc['content_view'].id:
-                        cvc_matched = _cvc
-                force_update = list()
-                if 'version' in component:
-                    cvc['version'] = find_content_view_version(module, cvc['content_view'], version=component['version'])
-                    cvc['latest'] = False
-                    if cvc_matched and cvc_matched.latest:
-                        # When changing to latest=False & version is the latest we must send 'content_view_version' to the server
-                        force_update.append('content_view_version')
-                if cvc_matched:
-                    cvc['composite_content_view'] = content_view_entity
-                    cvc_dict = sanitize_entity_dict(cvc, cvc_map)
-                    cvc_changed = naildown_entity_state(ContentViewComponent, cvc_dict, cvc_matched, 'present', module, force_update=force_update)
-                    current_cvcs.remove(cvc_matched)
-                    if cvc_changed:
-                        changed = cvc_changed
-                else:
-                    for attr in ['latest', 'version']:
-                        if attr not in cvc:
-                            cvc[attr] = None
-                    ContentViewComponent(composite_content_view=content_view_entity, content_view=cvc['content_view'],
-                                         latest=cvc['latest'], content_view_version=cvc['version']).add()
-                    changed = True
-        for cvc in current_cvcs:
+    # Also only act for composite content views
+    if (module.state == 'present' or (module.state == 'present_with_defaults' and changed)) and content_view_entity['composite'] and components is not None:
```

Comment:
Same here - consider using a boolean variable:

```suggestion
    composite_cv = (module.state == 'present' or (module.state == 'present_with_defaults' and changed)) and content_view_entity['composite'] and components is not None
    if compsite_cv:
```

Reply:
This is more a question of whether we need to update the list of components.

Reply:
I guess the comment isn't so clear, then... ;) `# Also only act for composite content views`

Reply:
Something to consider: when you ask for my review these days, I'm coming at the code as more of an outsider now that I've been less involved. It's not simple code to follow IMO for new contributors or old ones like me. It's these long methods or long one liner conditionals that really get me confused.

Reply:
Oh it goes together with `# only update CVC's of newly created or updated CCV's` (unchanged and above the changed block).
As a variable name, `update_components` come to my mind.

Reply:
> Something to consider: when you ask for my review these days, I'm coming at the code as more of an outsider now that I've been less involved. It's not simple code to follow IMO for new contributors or old ones like me. It's these long methods or long one liner conditionals that really get me confused.

That is what is making your review still very helpful, because being deeply involved like evgeni and me gives you blind spots on different levels.

Reply:
I'll split the boolean expression into two. Hopefully it's more readable.